rewrite-maven/src/main/java/org/openrewrite/maven/search/DependencyInsight.java

```diff
@@ -91,7 +91,7 @@ public String getDescription() {
                     ResolvedDependency dependency = findDependency(t);
```

Comment:
I think a better choice here would be to overload `findDependency()` on the MavenVisitor to have a two-argument variant that takes a tag and an optional scope. 

Then restrict the search to resolved dependencies in that scope.